### PR TITLE
Fix bug of removing URNs when updating fields looking up by URN

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

**Comment by nicpottier** [comment]
Hrmm.. so how would you remove a URN? Seems we need some way of indicating either removal, addition or setting.

**Comment by norkans7** [comment]
The bug here is that we actually remove while we should not

**Comment by nicpottier** [comment]
Well is that a bug? They are hitting an update contact endpoint and passing just one URN. So we update the contact to have just that one URN.

Whether that should be treated as an addition or a "setting" is ambiguity in the API. 

Same for groups and fields. What is our treatment there. @rowanseymour any opinions?

**Comment by norkans7** [comment]
Here there are looking up n the API by URN and they want to update the fields
but the issue is that we clear the existing URN and set the lookup urn only

I believe this was to be able to create new contacts but for existing contacts we should be keep all the existing URNs

File: temba/campaigns/tests.py [hunk 1]
```diff
@@ -1068,109 +1069,119 @@ def test_import_created_on_event(self):
...
-        # now update to the non-farmer group
-        self.nonfarmers = self.create_group("Not Farmers", [self.nonfarmer])
-        post_data = dict(name="Planting Reminders", group=self.nonfarmers.pk)
-        self.client.post(reverse("campaigns.campaign_update", args=[new_campaign.pk]), post_data)
-
-        # only one fire for the non-farmer the previous two should be deleted by the group change
-        self.assertEqual(1, EventFire.objects.filter(event__is_active=True).count())
-        self.assertEqual(2, EventFire.objects.filter(event__is_active=False).count())
-        self.assertEqual(1, EventFire.objects.filter(event__is_active=True, contact=self.nonfarmer).count())
+        with patch.object(timezone, "now", return_value=datetime(2020, 5, 1, 0, 0, 0, 0, pytz.UTC)):
```
Comment (norkans7) [inline temba/campaigns/tests.py:1072]: Just adding the mock since we are past "10-05-2020 12:30:10" in real life

**Comment by rowanseymour** [comment]
This looks right - the problem is specifically using a URN in the URL to lookup a contact and then that URN overwritting the contact's URNs

**Comment by nicpottier** [comment]
Oh, so this is about the lookup not an update? Ah, my bad!

**Pull request merged**